### DISCUSSION TITLE
Add ICDAR 2023 Datasets

### DISCUSSION THREAD

**Discussion by HamzaGbada** [mindee/doctr#1332] [Ideas]
**Summary:**
I would like to propose the addition of several datasets published in ICDAR 2023 to the `doctr` repository. These datasets are valuable resources for the document analysis and recognition community. Additionally, I'd like to suggest the inclusion of a new dataset named [WildReceipt](https://paperswithcode.com/paper/spatial-dual-modality-graph-reasoning-for-key) (Not published in ICDAR2023). This will enhance the repository's usefulness for researchers and practitioners working in this field.

**Datasets to be Added:**
1. DocVQA2020-23
2. ReST2023
3. SVRD2023
4. DSText2023
5. DUDE 😎2023
6. NewsVideoQA2023
7. RoadText2023
8. DocILE2023
9. WildReceipt

**Why These Datasets?**
ICDAR 2023 is a reputable conference in the field of document analysis and recognition. The datasets listed above are newly published and are crucial for advancing research and development in this area. They cover various aspects of document analysis, including text recognition, scene text detection, document layout analysis, and more.

**Additional Information:**
For more information about these datasets, check this [link](https://rrc.cvc.uab.es/?com=introduction)

What do you think @felixdittrich92 ?

**Comment by felixT2K** [comment]
Hi @HamzaGbada thanks a lot for the datasets list :)

I will check this tonight or maybe tomorrow morning 👍🏼 
Do you personally find any of the datasets particularly useful?

**Reply by HamzaGbada** [reply]
I've actually had some experience working with the WildReceipt dataset, and it's proven to be quite valuable for tasks related to receipt-based documents. However, I've encountered challenges with regard to data loading using the available APIs, including issues with the stability of the data loader, even when using the Hugging Face API. As for the other datasets you've mentioned, since they have been recently published, I haven't had the opportunity to work with them yet. Among them, I believe that [SVRD](https://rrc.cvc.uab.es/?ch=21) stands out as particularly intriguing and potentially valuable for our research and development efforts.

**Comment by felixT2K** [comment]
**1. DocVQA2020-23 - Looks interesting (contains Azure Document AI (OCR) annotations) ~13K samples**
2. ReST2023 - only non european texts
3. SVRD2023 - contains Chinese !?
4. DSText2023 - Video clips
5. DUDE 😎2023 - Looks like there are no annotations (only QA and Layout Understanding)
6. NewsVideoQA2023 - Video clips
7. RoadText2023 - Road Signs (Scene Text in the Wild)
8. DocILE2023 - only available with form request
**9. WildReceipt - https://huggingface.co/datasets/Theivaprakasham/wildreceipt  (Looks interesting)**

Additional:
**- https://github.com/google-research-datasets/hiertext**
**- https://github.com/microsoft/unilm/tree/master/textdiffuser#books-dataset   (enough data for training from scratch)**

I bolded the interesting ones :)

**Reply by HamzaGbada** [reply]
I agree with your assessments.

Regarding the implementation of the WildReceipt dataset, it's crucial to note that there is an issue with annotation. You can find more details about this issue in [this GitHub thread](https://github.com/huggingface/datasets/issues/6123#event-10120192184). This information is vital for anyone planning to work with the dataset.

**Reply by felixdittrich92** [reply]
Do you want to work on one or more dataset integrations in docTR ? :)

**Reply by HamzaGbada** [reply]
I would like to work on multiple dataset integrations in docTR! 😄

**Reply by felixT2K** [reply]
@HamzaGbada Nice sounds great 👍🏼 

You can take some inspiration from: https://github.com/mindee/doctr/pull/785/files  and https://github.com/mindee/doctr/blob/main/doctr/datasets/imgur5k.py (the read_from_folder function is only required if we cannot keep the box crops on RAM)

So which one do you like to add as first one ? :)
You can simply open a PR if you think it's fine don't hestiate if there are any open questions 😃

**Reply by HamzaGbada** [reply]
I'll begin with integrating the WildReceipt dataset. Now, a few questions to ensure a smooth integration process:

* Where should I specify the URL to download the dataset? Similar to this example:
```python
TRAIN = (
    "https://doctr-static.mindee.com/models?id=v0.1.1/sroie2019_train_task1.zip&src=0",
    "d4fa9e60abb03500d83299c845b9c87fd9c9430d1aeac96b83c5d0bb0ab27f6f",
    "sroie2019_train_task1.zip",
)
```
* Could you please guide me on where to find or how to implement the required tests before opening the pull request?
* Is there a specific deadline?

**Reply by felixT2K** [reply]
1. Looks like everything is bundled into one archive : https://download.openmmlab.com/mmocr/data/wildreceipt.tar ?
- So i would suggest to implement it with img and label folder (like IMGUR5K) in this case the user needs to download the dataset firstly and we provide only the dataset loader. So the license doesn't matter because we provide only the way to load the dataset and not the dataset directly.

2. We need to provide a mock of the dataset structure take a look at: https://github.com/mindee/doctr/blob/main/tests/conftest.py
Afterwards you can specify the test (in both test files `tests/pytorch/test_datasets_pt` and `tests/tensorflow/test_datasets_tf`  reference: https://github.com/mindee/doctr/blob/19f101cd300f9d978213e4c5d30944fafdbe3700/tests/pytorch/test_datasets_pt.py#L322

3. No not really :) We have planned the next release end of november / begin december

**Reply by HamzaGbada** [reply]
Thank you for the clarification. I'll plan to start the implementation process in two weeks since I have a conference to present at next week

**Reply by felixT2K** [reply]
Sure no stress 👍🏼 😃
